src/celeritas/grid/GenericGridInserter.hh

```diff
@@ -0,0 +1,136 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/grid/GenericGridInserter.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <utility>
+#include <vector>
+
+#include "corecel/Types.hh"
+#include "corecel/cont/Span.hh"
+#include "corecel/data/Collection.hh"
+#include "corecel/data/CollectionBuilder.hh"
+#include "celeritas/Types.hh"
+#include "celeritas/io/ImportPhysicsVector.hh"
+
+#include "GenericGridBuilder.hh"
+#include "GenericGridData.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct a generic grid using mutable host data and add it to
+ * the specified grid collection.
+ *
+ * \code
+    GenericGridInserter insert(&data->reals, &data->generic_grids);
+    std::vector<GenericGridIndex> grid_ids;
+    for (material : range(MaterialId{mats->size()}))
+        grid_ids.push_back(insert(physics_vector[material.get()]));
+   \endcode
+ */
+template<typename Index>
```

Comment:
Didn't notice this, but for style consistency we use "class" for the template argument, "typename" is only for resolving templates type aliases etc

Reply:
I only see a `GenericGridCalculator.test.cc` file -- I can add some tests there or also add make some `GenericGridBuilder` and `GenericGridInserter` like for the value grids

Reply:
Would you mind renaming `./celeritas/grid/GenericCalculator.test.cc` to `GenericGrid.test.cc` and adding it there?